### ISSUE TITLE
LGPL?

### ISSUE THREAD

**Issue by HarrisonM-SVL** [sstaub/Ethernet3#62]
@sstaub 

First off, thanks for the updated features in Ethernet3,  super useful!

For context, I'm working on a project using Ethernet3 that has the potential to be closed source, which is plausible with [Ethernet](https://github.com/arduino-libraries/Ethernet) and [Ethernet2](https://github.com/adafruit/Ethernet2) (with some limitations related to documentation and binary distribution) due to their being licensed with the [GNU Lesser Genera Public License](https://www.gnu.org/licenses/lgpl-3.0.html).

Per [Github's Documentation](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/licensing-a-repository#choosing-the-right-license):

"_You're under no obligation to choose a license. However, without a license, the default copyright laws apply, meaning that you retain all rights to your source code and no one may reproduce, distribute, or create derivative works from your work._"

**TLDR:** I think that if there is no license provided with this repository, you reserve all the rights, and it can't be modified or redistributed (open-source or closed) without your express permission.

Is it your intent for this library to be open-source? If this _isn't_ your intent, whatever, none of my business. But if this _is_ your intent, would you please consider selecting a license and adding it to this repository?

**Comment by sstaub** [comment]
The library is mainly based on Ethernet2. I forgot the license but it should the same as Ethernet2.
Many parts of the library have unknown authors and license types even in the Arduino library.
I will add it the next days.

**Comment by HarrisonM-SVL** [comment]
Much appreciated!

**Comment by HarrisonM-SVL** [comment]
@sstaub 

Clarifying question. You mentioned in your `licence.txt` that it contains

_"the gnu general public license, which covers the main body 
of the processing/arduino code (in general, all the stuff inside the 'app'
and 'core' subfolders)."_

And

_"the gnu lesser general public license that covers the arduino core
and libraries."_

Are both of these licenses intended to apply to everything within the `Ethernet3/src` folder of this repository? (i.e. Does the library contain some code that is licensed under GPL, and some that is under LGPL?)

Or is the GPL meant to apply to the Arduino Core (which I don't think is included in this repository), and the LGPL to the `Ethernet3` library?

I only ask because there are some differences between the 2 licenses, and I'm hashing out which one I need to use in a program that includes `Ethernet3`.

**Comment by sstaub** [comment]
Some parts of the library have different license types, e.g. EthernetUdp3 have MIT license.

**Comment by HarrisonM-SVL** [comment]
Gotcha, so you're saying some parts of the `Ethernet3` library have a GPL license, and some have a LGPL, which is why you put both?

**Comment by sstaub** [comment]
I take the license text from the Arduino library. I know it is chaotic but I can't change it.

**Comment by HarrisonM-SVL** [comment]
Thanks for your patience. When you say "Arduino Library" are you referring to older versions of the Ethernet Library such as those below?

https://github.com/adafruit/Ethernet2
https://github.com/arduino-libraries/Ethernet

**Comment by sstaub** [comment]
It is from Ethernet2

**Comment by HarrisonM-SVL** [comment]
Ok, per the `license.txt` you added to `Ethernet3`

_"next, the gnu lesser general public license that covers the arduino core
and libraries."_

Given that `Ethernet3` is a library, if I'm understanding the intent of your license correctly, I can assume that `Ethernet3` is a library licensed under the LGPL. Is that correct?
